secrets/secrets.go

```diff
@@ -111,8 +175,38 @@ func (k *Keeper) Encrypt(ctx context.Context, plaintext []byte) (ciphertext []by
 
 // Decrypt decrypts the ciphertext and returns the plaintext.
 func (k *Keeper) Decrypt(ctx context.Context, ciphertext []byte) (plaintext []byte, err error) {
-	ctx = k.tracer.Start(ctx, "Decrypt")
-	defer func() { k.tracer.End(ctx, err) }()
+	start := time.Now()
+	ctx, span := k.tracer.Start(ctx, "Decrypt")
+	// Set span attributes for testing
```

Comment:
So, it's not great that we went from 2 lines of code to record metrics to about 30. I suspect this will happen in every package. It's repetitive even inside this package (lines 130-160, lines 178-209 are basically the same).

a) Why does OpenTelemetry need so much more? For example, you're already passing the method name to the OpenSpan, why do we need to provide it again as an attribute?

b) Can you add a wrapper Start/End in the otel/ package that could be called here so that secrets/blob/pubsub/etc. only have to call Start and defer End?

Reply:
This is fixed now